pkg/client/apiutil/apimachinery.go

```diff
@@ -146,11 +151,29 @@ func GVKForObject(obj runtime.Object, scheme *runtime.Scheme) (schema.GroupVersi
 	return gvks[0], nil
 }
 
+// DefaultHTTPClient is a helper function that sets the HTTP client based on a rest config if it is not already set.
+func DefaultHTTPClient(config *rest.Config, httpClient **http.Client) error {
+	if httpClient == nil {
+		panic("httpClient must not be nil")
+	}
+
+	if *httpClient != nil {
+		return nil
+	}
+
+	var err error
+	*httpClient, err = rest.HTTPClientFor(config)
+	return err
```

Comment:
If we return a client and error here, instead of modifying the caller's pointer, we could reuse this function in a lot of other places where the defaulting logic is duplicated